collections/gamesCollection.mjs

```diff
@@ -45,21 +47,23 @@ class GamesCollection {
 
     url += "?format=json";
 
-    this.yf.api(this.yf.GET, url, (e, data) => {
-      if (e) {
-        return cb(e);
-      }
-
-      const games = parseCollection(data.fantasy_content.games, subresources);
-      return cb(null, games);
-    });
+    return this.yf.api(this.yf.GET, url)
+      .then(data => parseCollection(data.fantasy_content.games, subresources))
+      .then(games => { 
+        cb(null, games); 
+        return games; 
```

Comment:
took me a second to realize why these `return`s were needed. I see it now... clever!

Reply:
... would it make sense to execute the callback in the previous block? curious as i know promises in nodejs are not super performant...

Reply:
(i should say, I know promises are slower than callbacks, but I guess I need to figure out if chaining `then`s has much of an impact on performance...

Reply:
What would the "previous block" be? the `then(...) in line 51 to reduce the chaining by one level? Sure.

If you are concerned with the performance overhead, the option offered in https://github.com/whatadewitt/yfsapi/pull/45 has no chaining at all, while this one does require chaining in every method.